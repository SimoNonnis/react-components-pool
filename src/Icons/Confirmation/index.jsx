import React from 'react';
import PropTypes from 'prop-types';

function Confirmation({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 272 272"
      width={width}
      height={height}
      className={className}
    >
      <title>Confirmation Icon</title>

      <path
        className={classNamePath}
        fill={fill}
        d="M136,0A136,136,0,1,0,272,136,136,136,0,0,0,136,0ZM122.46,211,46,134.54l25.46-25.46L120.38,158l80.36-94.06,27.37,23.38Z"
      />
    </svg>
  );
}

Confirmation.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

Confirmation.defaultProps = {
  width: 40,
  height: 40,
  fill: 'currentColor',
  className: 'sizeIcon',
  classNamePath: 'sizeIconPath',
};

export default Confirmation;
