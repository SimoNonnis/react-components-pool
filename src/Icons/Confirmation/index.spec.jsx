/* eslint-env jest */
import React from 'react';

import Confirmation from './';

describe('Test <Confirmation /> icon', () => {
  it('should render Confirmation icon correctly with default props', () => {
    const wrapper = shallow(<Confirmation />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Confirmation icon correctly with props', () => {
    const wrapper = shallow(
      <Confirmation fill="#a4a1a1" className="testClassname" height={30} width={30} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
