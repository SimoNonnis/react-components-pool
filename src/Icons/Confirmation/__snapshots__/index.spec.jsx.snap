// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Confirmation /> icon should render Confirmation icon correctly with default props 1`] = `
<svg
  className="sizeIcon"
  height={40}
  viewBox="0 0 272 272"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Confirmation Icon
  </title>
  <path
    className="sizeIconPath"
    d="M136,0A136,136,0,1,0,272,136,136,136,0,0,0,136,0ZM122.46,211,46,134.54l25.46-25.46L120.38,158l80.36-94.06,27.37,23.38Z"
    fill="currentColor"
  />
</svg>
`;

exports[`Test <Confirmation /> icon should render Confirmation icon correctly with props 1`] = `
<svg
  className="testClassname"
  height={30}
  viewBox="0 0 272 272"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Confirmation Icon
  </title>
  <path
    className="sizeIconPath"
    d="M136,0A136,136,0,1,0,272,136,136,136,0,0,0,136,0ZM122.46,211,46,134.54l25.46-25.46L120.38,158l80.36-94.06,27.37,23.38Z"
    fill="#a4a1a1"
  />
</svg>
`;
